### PR TITLE
AilurusUmbra proposal

### PR DESCRIPTION
**Pull request by AilurusUmbra**
This PR includes following files:

* `README.md`: proposal of [EEGNN](https://github.com/AilurusUmbra/EEGNN)

---
A good PR should include the following parts:

- [x] A clear title
- [x] A meaningful message for PR, as well as its commits
- [x] from your specific branch (***not master***) merging to teacher's `HW` branch (***not master***, again)
- [x] excluding any irrelevant files, such as binaries, text files, or dot files
- [ ] passing tests/CI

### CODE REVIEW

File: proposal/AilurusUmbra/README.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+ 
+
+## Perspective users
+
+* BCI researchers for processing EEG/MEG.
+* Engineers that need to do multichannel time series prediction/classification.
+
+## System architecture
+
+Most of the computation would be leverage to Pytorch framework, so that tensors could be easily move to VRAM of GPU to compute, and move back to RAM after computation.
```
Comment (yungyuc) [inline proposal/AilurusUmbra/README.md:81]: The course expects you to write C++ code on CPU.  If you write GPU code, or use only Pytorch Python API, your do not implement C++/Python interface yourselves, and do not fulfill the expectation.

You can try to address the omission, but I am not sure what is the best way to do it.  If the omission remains unaddressed at the end of the project, your grade will be affected.

File: proposal/AilurusUmbra/README.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    * models.skipinit()
+
+* Utils: Some supplementary functions could just use api of [mne](https://mne.tools/stable/index.html) to prevent from reinventing the wheel. For example, read FIF data (MEG) or load sample dataset.
+
+  * mne.io.read_raw_fif(fname[, allow_maxshield, …])            Reader function for Raw FIF data.
+  * mne.datasets.eegbci.load_data(subject, runs[, path, …])     Get paths to local EEGBCI dataset files.
+
+## Engineering infrastructure
+
+This project would be mainly developed by python>=3.6 and based on Pytorch>=1.0 and tested on Debian, Mac OS and Win10 for compatibility. If training models from scratch is needed, it is recommended that running EEGNN on nvidia GPUs with at least 6GB VRAM (e.g. GTX 1060) for less training time. For only fine-tuning pretrained models or just doing inference, EEGNN provides multi-thread mechanism for CPU computation speedup. EEGNN is designed for both high performance computing on training models or efficient inference on BCI portable devices.
```
Comment (yungyuc) [inline proposal/AilurusUmbra/README.md]: Validating and testing should be included in your engineering infrastructure.

Comment (AilurusUmbra) [reply proposal/AilurusUmbra/README.md]: This is modified in this commit [d96a523](https://github.com/yungyuc/nsdhw_20sp/pull/67/commits/d96a523368017530f4807ab4acb4247de3fe1cd8)

File: proposal/AilurusUmbra/README.md (hunk 1 above)
Comment (AilurusUmbra) [reply proposal/AilurusUmbra/README.md:81]: I think I should take some time to change this proposal into mainly C++ project (maybe compatible with PyTorch). I would commit this later.

Comment (AilurusUmbra) [reply proposal/AilurusUmbra/README.md:81]: I remove the GPU part in this commit [e07643d](https://github.com/AilurusUmbra/nsdhw_20sp/commit/e07643d3e677a07449128c375017a2c6f70c133c)

**Pull request merged**